src/stream/src/common/log_store/kv_log_store/buffer.rs

```diff
@@ -39,8 +48,12 @@ pub(crate) enum LogStoreBufferItem {
 }
 
 struct LogStoreBufferInner {
-    queue: VecDeque<LogStoreBufferItem>,
+    /// Items not read by log reader. Newer item at the front
+    unconsumed_queue: VecDeque<(u64, LogStoreBufferItem)>,
+    /// Items already read by log reader by not truncated. Newer item at the front
+    consumed_queue: VecDeque<(u64, LogStoreBufferItem)>,
```

Comment:
Correct me if I am wrong, `unconsumed_queue` and `consumed_queue` are introduced for future optimization and technically not useful for now. In the future, we can put items that are persisted in sink in `consumed_queue` and don't need to flush items in `consumed_queue` when checkpointing.

Reply:
They are currently in use. Items in `consumed_queue` means the buffer item has been emitted to the log reader, while the log reader has not truncated it yet. We have to store these items because when checkpoint barrier comes, we have to flush the stream chunk in the `consumed_queue`, since the log reader has not truncated them and these stream chunks are not persisted in the sink yet.

Reply:
In current code we don't have a flushed offset yet. Because the stream chunk can be added to the buffer whenever there is available quota. Therefore the flushed stream chunk will be scattered and is not continuous. This is the reason why we maintain a `is_flushed` flag for each stream chunk.